src/Microsoft.Identity.Web.DownstreamApi/DownstreamApi.cs

```diff
@@ -286,15 +287,45 @@ internal partial class DownstreamApi : IDownstreamApi
             // Downstream API URI
             string apiUrl = effectiveOptions.GetApiUrl();
 
-            // Creation of the Http Request message with the right HTTP Method
-            using HttpRequestMessage httpRequestMessage = new HttpRequestMessage(
+            using HttpClient client = string.IsNullOrEmpty(serviceName) ? _httpClientFactory.CreateClient() : _httpClientFactory.CreateClient(serviceName);
+
+            // Create an HTTP request message
+            using HttpRequestMessage httpRequestMessage = new(
                 new HttpMethod(effectiveOptions.HttpMethod),
                 apiUrl);
-            if (content != null)
-            {
-                httpRequestMessage.Content = content;
-            }
 
+            await UpdateRequestAsync(httpRequestMessage, content, effectiveOptions, appToken, user, cancellationToken);
+
+            // Send the HTTP message           
+            var downstreamApiResult = await client.SendAsync(httpRequestMessage, cancellationToken).ConfigureAwait(false);
+
+            effectiveOptions.AcquireTokenOptions.Claims = WwwAuthenticateParameters.GetClaimChallengeFromResponseHeaders(downstreamApiResult.Headers);

```

Comment:
Why do we get the Claims if we don't have a 401? can it happen that there is a claims challenge with a 200? What is the scenario?

Apparently we only retry in the case of a 401 below. So I think we should delay extracting the claims until we have tested it's a 401?